packages/components/src/internal/actions.ts

```diff
@@ -1097,6 +1098,48 @@ export function setSelected(
     });
 }
 
+/**
+ * Selects individual ids for a particular view
+ * @param key the selection key for the grid
+ * @param ids ids to change selection for
+ * @param containerPath optional path to the container for this grid.  Default is the current container path
+ */
+export function replaceSelected(
```

Comment:
If possible, ComponentWithQueryModels replaceSelections action should be updated to use this function